openedx/core/lib/api/tests/test_authentication.py

```diff
@@ -111,51 +125,78 @@ def setUp(self):
+        auth = self._create_authorization_header(token)
+        return self.csrf_client.post(target_url, HTTP_AUTHORIZATION=auth)
+
+    def check_error_codes(self, response, status_code, error_code):
+        """
+        Ensure that the response has the appropriate HTTP status, and provides
+        the expected error_code in the JSON response body.
+        """
+        response_dict = json.loads(response.content)
+        self.assertEqual(response.status_code, status_code)
+        self.assertEqual(response_dict['error_code'], error_code)
+
     def _create_authorization_header(self, token=None):  # pylint: disable=missing-docstring
-        return "Bearer {0}".format(token or self.access_token.token)
+        if token is None:
+            token = self.access_token.token
+        return "Bearer {0}".format(token)
 
+    @ddt.data(None, {})
     @unittest.skipUnless(oauth2_provider, 'django-oauth2-provider not installed')
-    def test_get_form_with_wrong_authorization_header_token_type_failing(self):
+    def test_get_form_with_wrong_authorization_header_token_type_failing(self, params):
         """Ensure that a wrong token type lead to the correct HTTP error status code"""
-        auth = "Wrong token-type-obviously"
-        response = self.csrf_client.get('/oauth2-test/', {}, HTTP_AUTHORIZATION=auth)
-        self.assertEqual(response.status_code, 401)
-        response = self.csrf_client.get('/oauth2-test/', HTTP_AUTHORIZATION=auth)
-        self.assertEqual(response.status_code, 401)
-
-    @unittest.skipUnless(oauth2_provider, 'django-oauth2-provider not installed')
-    def test_get_form_with_wrong_authorization_header_token_format_failing(self):
-        """Ensure that a wrong token format lead to the correct HTTP error status code"""
-        auth = "Bearer wrong token format"
-        response = self.csrf_client.get('/oauth2-test/', {}, HTTP_AUTHORIZATION=auth)
-        self.assertEqual(response.status_code, 401)
-        response = self.csrf_client.get('/oauth2-test/', HTTP_AUTHORIZATION=auth)
-        self.assertEqual(response.status_code, 401)
-
-    @unittest.skipUnless(oauth2_provider, 'django-oauth2-provider not installed')
-    def test_get_form_with_wrong_authorization_header_token_failing(self):
-        """Ensure that a wrong token lead to the correct HTTP error status code"""
-        auth = "Bearer wrong-token"
-        response = self.csrf_client.get('/oauth2-test/', {}, HTTP_AUTHORIZATION=auth)
-        self.assertEqual(response.status_code, 401)
-        response = self.csrf_client.get('/oauth2-test/', HTTP_AUTHORIZATION=auth)
-        self.assertEqual(response.status_code, 401)
-
+        response = self.csrf_client.get(
+            '/oauth2-test/',
+            params,
+            HTTP_AUTHORIZATION='Wrong token-type-obviously'
+        )
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+        # If no bearer token is provided, authorization passes to the next registered
```

Comment:
This confused me at first.  By "no bearer token" here, you mean, no authorization header with the Bearer prefix.  As opposed to no token provided with the Bearer authorization (which is tested below).  Right?